project/types.go

```diff
@@ -237,6 +237,7 @@ type Service interface {
 	DependentServices() []ServiceRelationship
 	Containers() ([]Container, error)
 	Scale(count int) error
+	SpecificiesHostPort() bool
```

Comment:
I would definitely removed this from the interface.  This is a property of the Docker implementation and no the definition of a Service itself.